src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,153 @@
+        String[] commandComponents = fullCommand.split(" ");
+        String command = commandComponents[0];
+
+        // Command arguments
+        int task;
+        String description;
+        LocalDate by;
+        String at;
+        String arguments;
+        String[] argumentComponents;
+        String keyword;
+
+        switch (command) {
+        case Parser.DONE_COMMAND:
+            // Check for missing task argument
+            if (commandComponents.length == 1) {
+                throw new MissingArgumentException("Done", "task");
+            }
+            // Extract task argument
+            task = Integer.parseInt(commandComponents[1]) - 1;
+            // Check for invalid task argument
+            if (task < 0) {
+                throw new InvalidArgumentException(task);
+            }
+            // If pass all checking, create and return done command obj
+            return new DoneCommand(task);
+
+        case Parser.DELETE_COMMAND:
+            // Check for missing task argument
+            if (commandComponents.length == 1) {
+                throw new MissingArgumentException("Delete", "task");
+            }
+            // Extract task argument
+            task = Integer.parseInt(commandComponents[1]) - 1;
+            // Check for invalid task argument
+            if (task < 0) {
+                throw new InvalidArgumentException(task);
+            }
+            // If pass all checking, create and return delete command obj
+            return new DeleteCommand(task);
+
+        case Parser.FIND_COMMAND:
+            // Check for missing keyword argument
+            if (commandComponents.length == 1) {
+                throw new MissingArgumentException("Find", "keyword");
+            }
+            // Extract keyword argument
+            keyword = commandComponents[1];
+            // If pass all checking, create and return find command obj
+            return new FindCommand(keyword);
+
+        case Parser.DEADLINE_COMMAND:
+            // Check for empty description
+            if (commandComponents.length == 1) {
+                throw new EmptyDescriptionException("Deadline");
+            }
+            arguments = fullCommand.split(" ", 2)[1];
+            argumentComponents = arguments.split(" /by ");
+            // Extract description argument
+            description = argumentComponents[0];
+            // Check for missing by argument
+            if (argumentComponents.length == 1) {
+                throw new MissingArgumentException("Deadline", "/by");
+            }
+            // Extract and parse by argument
+            try {
+                by = LocalDate.parse(arguments.split(" /by ")[1]);
+            } catch (DateTimeParseException err) {
+                throw new InvalidDateFormatException();
+            }
+            // If pass all checking, create and return deadline command obj
+            return new DeadlineCommand(description, by);
```

Comment:
I really like the comments, they help in understanding the code! However, would it be better to leave several line breaks here as well to improve readability? The other `case` blocks can also be reviewed to see if the addition of line breaks would improve readability.

Reply:
Glad the comments help! Yes, I shall add in the line breaks.